src/app/map/search/search.component.ts

```diff
@@ -0,0 +1,156 @@
+import { Component, Input, Output, EventEmitter, SimpleChanges, OnDestroy, OnChanges, OnInit } from '@angular/core';
+import { Location } from '@angular/common';
+import { Observable } from 'rxjs/Observable';
+import { Subject } from 'rxjs/Subject';
+import { ActivatedRoute, Router, Params } from '@angular/router';
+import { Project } from 'app/models/project';
+import 'rxjs/add/operator/debounceTime';
+import 'rxjs/add/operator/distinctUntilChanged';
+import 'rxjs/add/operator/map';
+import * as _ from 'lodash';
+
+export interface FiltersType {
+  mineFilter: string;
+  permitFilter: string;
+}
+
+@Component({
+  selector: 'map-search',
+  templateUrl: './search.component.html',
+  styleUrls: ['./search.component.scss']
+})
+export class SearchComponent implements OnInit, OnChanges, OnDestroy {
+
+  @Input() projects: Array<Project> = [];
+  @Input() selected: string;
+  @Output() updateMatching = new EventEmitter();
+
+  public loading = false;
+  public ranSearch = false;
+  public mineFilter: string = null;
+  public _mineFilter: string = null; // temporary filters for Cancel feature
+  public permitFilter: string = null;
+  public _permitFilter: string = null; // temporary filters for Cancel feaure
+  public typeahead: Observable<string> = null;
+  public resultsCount = 0;
+
+  public radioSel: string;
+  public radioOptions: string[] = ['Mine Name', 'Permit Number'];
+  private mineKeys: Array<string> = [];
+  private permitKeys: Array<string> = [];
+
+  private ngUnsubscribe: Subject<boolean> = new Subject<boolean>();
+
+  constructor(
+    private location: Location,
+    private route: ActivatedRoute,
+    private router: Router,
+  ) {}
+
+  // (arrow) functions to return type-ahead results
+  // ref: https://ng-bootstrap.github.io/#/components/typeahead/api
+  //
+  public mineSearch = (text$: Observable<string>) =>
+    text$
+      .debounceTime(200)
+      .distinctUntilChanged()
+      .map(term => term.length < 1 ? []
+        : this.mineKeys.filter(key => key.indexOf(this._mineFilter.toUpperCase()) > -1) // .slice(0, 10)
```

Comment:
Can the slice be removed?

Reply:
yep, missed cleaning it with other comments